docs/ovms_onnx_example.md

```diff
@@ -79,11 +84,86 @@ output = make_ndarray(result.outputs["gpu_0/softmax_1"])
 max = np.argmax(output)
 print("Class is with highest score: {}".format(max))
 print("Detected class name: {}".format(classes.imagenet_classes[max]))
-
 ```
+
 It shows the following output:
 ```bash
 bee.jpeg (1, 3, 224, 224) ; data range: -2.117904 : 2.64
 Class is with highest score: 309
 Detected class name: bee
-```
\ No newline at end of file
+```
+
+
+## Option 2: Adding preprocessing to the server side (building a DAG) <a name="#server-side"></a>
+
+Create configuration file with DAG containing two sequential nodes: one being the _image transformation node_ and one _DL model node_ resnet. The job of image transformation node will be to preprocess the image data to match format required by ONNX model `resnet50-caffe2-v1-9.onnx`.
+
+The example [configuration file](../src/custom_nodes/image_transformation/config_with_preprocessing_node.json) is available in _image transformation_ custom node directory.  
+Image transformation custom node library building steps can be found [here](../src/custom_nodes/image_transformation).
+
+Prepare workspace with the model, preprocessing node library and configuration file.
+```
+$ tree workspace
+
+workspace
+├── config_with_preprocessing_node.json
+├── lib
+│   └── libcustom_node_image_transformation.so
+└── models
+    └── resnet50-caffe2-v1
+        └── 1
+            └── resnet50-caffe2-v1-9.onnx
+```
+
+Start the OVMS container with configuration file option:
+```bash
+docker run -d -u $(id -u):$(id -g) -v $(pwd)/resnet:/model -p 9001:9001 openvino/model_server:latest \
+--config_path /workspace/config_with_preprocessing_node.json --port 9001
+```
+
+Use example binary input client to send JPEG images to the server:
+```
+$ cd model_server/example_client
+
+$ python3 grpc_binary_client.py --grpc_port 9001 --input_name 0 --output_name 1463 --model_name resnet --batchsize 1
+```
+You should see 100% overall accurracy:
+```
+Start processing:
+        Model name: resnet
+        Images list file: input_images.txt
+Batch: 0; Processing time: 38.65 ms; speed 25.87 fps
```

Comment:
maybe the performance would be better with 1 stream on ovms side?

Reply:
There is significant perf boosst with 1 stream. Changed the doc and config file.